Frameworks/Foundation/NSObject_NSKeyValueArrayAdapter.mm

```diff
@@ -62,265 +53,15 @@
-        [_get() getObjects:objects range:range];
-    }
-
-    /* Mutables */
-    void insertObjectAtIndex(id object, NSUInteger index) {
-        if (_targetSelectors.insertAtOne) {
-            _call<void>(_targetSelectors.insertAtOne, object, index);
-            return;
-        }
-
-        NSIndexSet* indexes = [NSIndexSet indexSetWithIndex:index];
-        [_target willChange:NSKeyValueChangeInsertion valuesAtIndexes:indexes forKey:_key];
-
-        _getMutateAndSet(@selector(insertObject:atIndex:), object, index);
-
-        [_target didChange:NSKeyValueChangeInsertion valuesAtIndexes:indexes forKey:_key];
-    }
-
-    void removeObjectAtIndex(NSUInteger index) {
-        if (_targetSelectors.removeAtOne) {
-            _call<void>(_targetSelectors.removeAtOne, index);
-            return;
-        }
-
-        NSIndexSet* indexes = [NSIndexSet indexSetWithIndex:index];
-        [_target willChange:NSKeyValueChangeRemoval valuesAtIndexes:indexes forKey:_key];
-
-        _getMutateAndSet(@selector(removeObjectAtIndex:), index);
-
-        [_target didChange:NSKeyValueChangeRemoval valuesAtIndexes:indexes forKey:_key];
-    }
-
-    void addObject(id object) {
-        if (_targetSelectors.insertAtOne) {
-            insertObjectAtIndex(object, count());
-            return;
-        }
-
-        NSIndexSet* indexes = [NSIndexSet indexSetWithIndex:count()];
-        [_target willChange:NSKeyValueChangeInsertion valuesAtIndexes:indexes forKey:_key];
-
-        _getMutateAndSet(@selector(addObject:), object);
-
-        [_target didChange:NSKeyValueChangeInsertion valuesAtIndexes:indexes forKey:_key];
-    }
-
-    void removeLastObject() {
-        if (_targetSelectors.removeAtOne) {
-            removeObjectAtIndex(count() - 1);
-            return;
-        }
-
-        NSIndexSet* indexes = [NSIndexSet indexSetWithIndex:count() - 1];
-        [_target willChange:NSKeyValueChangeRemoval valuesAtIndexes:indexes forKey:_key];
-
-        _getMutateAndSet(@selector(removeLastObject));
-
-        [_target didChange:NSKeyValueChangeRemoval valuesAtIndexes:indexes forKey:_key];
-    }
-
-    void replaceObjectAtIndexWithObject(NSUInteger index, id object) {
-        if (_targetSelectors.replaceOne) {
-            _call<void>(_targetSelectors.replaceOne, index, object);
-            return;
-        } else if (_targetSelectors.removeAtOne && _targetSelectors.insertAtOne) {
-            _call<void>(_targetSelectors.removeAtOne, index);
-            _call<void>(_targetSelectors.insertAtOne, object, index);
-            return;
-        }
-
-        NSIndexSet* indexes = [NSIndexSet indexSetWithIndex:count() - 1];
-        [_target willChange:NSKeyValueChangeReplacement valuesAtIndexes:indexes forKey:_key];
-
-        _getMutateAndSet(@selector(replaceObjectAtIndex:withObject:), index, object);
-
-        [_target didChange:NSKeyValueChangeReplacement valuesAtIndexes:indexes forKey:_key];
-    }
-};
-}
-
 @interface _NSKeyProxyArray () {
```

Comment:
nit: it's strange that the filenames differ so much from the actual class names...